src/openklant/components/klantinteracties/api/serializers/klantcontacten.py

```diff
@@ -537,3 +537,59 @@ def create(self, validated_data):
         )
 
         return super().create(validated_data)
+
+
+class BetrokkeneConvenienceSerializer(BetrokkeneSerializer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["had_klantcontact"].read_only = True
+
+
+class OnderwerpobjectConvenienceSerializer(OnderwerpobjectSerializer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["klantcontact"].read_only = True
+
+
+class KlantContactBetrokkeneOnderwerpObjectSerializer(serializers.Serializer):
+    klantcontact = KlantcontactSerializer()
+    betrokkene = BetrokkeneConvenienceSerializer(required=False)
+    onderwerpobject = OnderwerpobjectConvenienceSerializer(required=False)
+
+    @transaction.atomic
+    def create(self, validated_data):
+        """
+        Create the objects and use the original serializers to ensure all the correct
+        fields show up in the response
+        """
+        klantcontact_data = validated_data["klantcontact"]
+        klantcontact = Klantcontact.objects.create(**klantcontact_data)
+
+        betrokkene = None
+        if betrokkene_data := validated_data.pop("betrokkene", None):
+            betrokkene_data["had_klantcontact"] = {"uuid": str(klantcontact.uuid)}
+            # TODO for some reason `was_partij` is converted to `partij` by the serializer
```

Comment:
It is caused by the `source` option given to the `was_partij` field https://github.com/maykinmedia/open-klant/blob/2963f4c2b73e0e24759b24eb8e5f8469066ccafe/src/openklant/components/klantinteracties/api/serializers/klantcontacten.py#L139

Reply:
Hmm I see, I don't think there's a way around this, since it always gets deserialized to `partij` right?

Reply:
Yeah, I never knew this was how the `source` option behaved :open_mouth:. The comment can be removed though right?